### DISCUSSION TITLE
React useQuery returns data from cache with "network-only"

### DISCUSSION THREAD

**Discussion by govorukhin-do** [urql-graphql/urql#2784] [Q&A] [answered]
### Describe the bug

Hello. In my project there is a "transactions" query with relay pagination. 
cacheExchange resolvers:
```
Query: {
    transactions: relayPagination(),
}
```

I use data returned from useQuery with  requestPolicy: "network-only". And loading 10 items per request (passing "first" and "after" arguments)

```
const [{ data, fetching, error }] = useQuery({
    query,
    variables: { first: 10 },
    requestPolicy:  "network-only",
    pause
});
```
Next pages I load through  graphqlClient returned from `useClient()`
And `data` from  `useQuery()`  is updated after completion `graphqlClient.query(...)`

For example, scenario with unexpected behaviour:
1. I go to transactions page and load two pages sequentially, after that  `data.transactions.edges` return 20 items, its ok.
2. I go to another page, so component with useQuery was unmount.
3. I go to transactions page again. useQuery starts new request for first page, arguments ` { first: 10 }`, its still ok.
4. But `data.transactions.edges ` will still return 20 items. But I expected 10 items when  "network-only" used.

Docs says that "_network-only will always send an API request and will ignore cached results."_

Сan you give me a hint how to get the expected behavior? Or is it a bug?

### Reproduction

https://codesandbox.io/s/solitary-bird-ggf8cq

### Urql version

2.2.1

### Validations

- [X] I can confirm that this is a bug report, and not a feature request, RFC, question, or discussion, for which [GitHub Discussions](https://github.com/FormidableLabs/urql/discussions) should be used
- [X] Read the [docs](https://formidable.com/open-source/urql/docs/).
- [X] Follow our [Code of Conduct](https://github.com/FormidableLabs/urql/blob/main/CODE_OF_CONDUCT.md)

**Comment by kitten** [comment, marked as answer]
I can convert this in a discussion in a bit, unless we find a resolution quickly, but this looks more like a question to me.

In short, just because you set `network-only`, that doesn't mean that Graphcache is fully bypassed. In fact, request policies dictate when a request should be made, but not how the response is handled. If that was the case, then any response from the GraphQL schema wouldn't run through resolvers and you'd get inconcistent results.

So to apply that to your specific case, the `relayPagination` helper is a quick and easy way to integrate Relay Pages, but `network-only` _does not_ mean that `resolvers` are bypassed. In fact, the resolvers are still applied to your API result, and triggering a `network-only` operation only means that you'll get a result from the API rather than the cache (you're telling the cache to immediately disregard its data and get an API result)

That API result is still however:
- written to the cache
- resolved against the cache
- `updates` and then `resolvers` will be run

Personally, if you're looking for a full solution, that's why we tell people when asked that _infinite pagination_ is a _UI-concern_. Your UI basically displays multiple pages, and `relayPagination` is just a quick and dirty way for the cache to take over that concern.

That's why we've got a separate non-normalised-cache pattern here: https://formidable.com/open-source/urql/docs/basics/ui-patterns/#infinite-scrolling

However, otherwise, be aware that `resolvers` always apply for the sake of consistency to API results as well, so that they actually consistently apply their changes and transformations to your GraphQL data.

**Comment by govorukhin-do** [comment]
Thanks for the answer. Well, I understand your point of view, this issue can be transformed into a discussion
